src/Http/Server.php

```diff
@@ -13,343 +13,26 @@
-     *
-     * @return string
-     */
-    public static function address(): string
-    {
-        return static::get('SERVER_ADDR', '');
-    }
-
-    /**
-     * Checks if the request is being served by the CLI
-     *
-     * @return bool
-     */
-    public static function cli(): bool
-    {
-        if (static::$cli !== null) {
-            return static::$cli;
-        }
-
-        if (defined('STDIN') === true) {
-            return static::$cli = true;
-        }
-
-        $term = getenv('TERM');
-
-        if (substr(PHP_SAPI, 0, 3) === 'cgi' && $term && $term !== 'unknown') {
-            return static::$cli = true;
-        }
-
-        return static::$cli = false;
-    }
-
-    /**
-     * Gets a value from the _SERVER array
-     *
-     * <code>
-     * Server::get('document_root');
-     * // sample output: /var/www/kirby
-     *
-     * Server::get();
-     * // returns the whole server array
-     * </code>
-     *
-     * @param mixed $key The key to look for. Pass false or null to
-     *                   return the entire server array.
-     * @param mixed $default Optional default value, which should be
-     *                       returned if no element has been found
-     * @return mixed
-     */
-    public static function get($key = null, $default = null)
-    {
-        if ($key === null) {
-            return $_SERVER;
-        }
-
-        $key   = strtoupper($key);
-        $value = $_SERVER[$key] ?? $default;
-        return static::sanitize($key, $value);
-    }
-
-    /**
-     * Returns the correct host
-     *
-     * @param bool $forwarded Deprecated. Todo: remove in 3.7.0
-     * @return string
-     */
-    public static function host(bool $forwarded = false): string
-    {
-        $hosts[] = static::get('SERVER_NAME');
-        $hosts[] = static::get('SERVER_ADDR');
-
-        // insecure host parameters are only allowed when hosts
-        // are validated against set of host patterns
-        if (empty(static::$hosts) === false) {
-            $hosts[] = static::get('HTTP_HOST');
-            $hosts[] = static::get('HTTP_X_FORWARDED_HOST');
-        }
-
-        // remove empty hosts
-        $hosts = array_filter($hosts);
-
-        foreach ($hosts as $host) {
-            if (static::isAllowedHost($host) === true) {
-                return explode(':', $host)[0];
-            }
-        }
-
-        return '';
-    }
-
-    /**
-     * Setter and getter for the the static $hosts property
-     *
-     * $hosts = null                     -> return all defined hosts
-     * $hosts = Server::HOST_FROM_SERVER -> []
-     * $hosts = Server::HOST_FROM_HEADER -> ['*']
-     * $hosts = array                    -> [array of trusted hosts]
-     * $hosts = string                   -> [single trusted host]
-     *
-     * @param string|array|int|null $hosts
-     * @return array
+     * @return \Kirby\Http\Environment
      */
-    public static function hosts($hosts = null): array
```

Comment:
This method was removed in the new implementation as it's no longer needed by the core. But could it be called by users or plugins? Do we need it for compatibility until the `Server` class gets removed in 3.8?

Reply:
We introduced it in 3.6.x somewhere. It's super young. I think it would be fine to remove it already again as a breaking change. It's very unlikely that someone already started using it within the last few months. It was hardly documented at all.

Reply:
Agreed, it's a very young internal method, I think this will do no harm removing it now as a breaking change.

Reply:
I've added it to the release note list in the first post.